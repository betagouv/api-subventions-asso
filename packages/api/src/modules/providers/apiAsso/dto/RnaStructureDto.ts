export interface RnaStructureDto {
    identite: {
        nom: string;
        sigle: string;
        id_rna: string;
        date_pub_jo: string;
        date_modif_rna: string;
        active: boolean;
        nature: string;
    };
    activites: {
        objet: string;
        lib_objet_social1: string;
        annee_activite_principale: number;
        effectif_salarie_cent: number;
        annee_effectif_salarie_cent: number;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            cp: number;
            bp?: string;
            type_voie: string;
        };
        adresse_gestion: {
            cplt_1: string;
            voie: string;
            commune: string;
            cp: number;
            pays: string;
        };
    };
}
