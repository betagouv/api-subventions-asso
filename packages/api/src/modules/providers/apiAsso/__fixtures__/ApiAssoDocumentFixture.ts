import { StructureDacDocumentDto, StructureRnaDocumentDto } from "../dto/StructureDto";

export const ApiAssoDocumentFixture = {
    asso: {
        documents: {
            nbDocDac: 1,
            document_dac: [
                {
                    url: "/fake/url",
                    nom: "nom fake",
                    meta: {
                        type: "BPA",
                        id_siret: 50922194100000,
                        etat: "courant",
                    },
                    time_depot: "2021-06-18 12:02:53",
                },
            ] as StructureDacDocumentDto[],
            nbDocRna: 1,
            document_rna: [
                {
                    url: "/fake/url",
                    type: "PV",
                    sous_type: "PV",
                    id: "ididididid",
                    lib_sous_type: "PV",
                    time: 1622557171,
                    annee: 2021,
                },
            ] as StructureRnaDocumentDto[],
        },
    },
};
