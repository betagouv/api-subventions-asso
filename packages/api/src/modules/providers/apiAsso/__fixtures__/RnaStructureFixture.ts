import { RnaStructureDto } from "../dto/RnaStructureDto";

export const rnaStructureFixture: RnaStructureDto = {
    identite: {
        nom: "Asso qui n'existe pas",
        sigle: "Asso qui n'existe p...",
        id_rna: "W000000000",
        date_pub_jo: "1998-02-07",
        date_modif_rna: "2023-01-12",
        active: true,
        nature: "Simplement déclarée"
    },
    activites: {
        objet: "Object social !",
        lib_objet_social1: "conduite d'activités économiques",
        annee_activite_principale: 0,
        effectif_salarie_cent: 0,
        annee_effectif_salarie_cent: 0
    },
    coordonnees: {
        adresse_siege: {
            cplt_1: "Place des orties",
            num_voie: "18",
            voie: "MEURIS",
            commune: "Londre",
            cp: "42424",
            type_voie: "RUE"
        },
        adresse_gestion: {
            cplt_1: "Place des orties",
            voie: "MEURIS",
            commune: "Londre",
            cp: "42424",
            pays: "Groenland"
        }
    }
};
