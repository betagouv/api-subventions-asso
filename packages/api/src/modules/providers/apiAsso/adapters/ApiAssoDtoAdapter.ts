import { siretToNIC } from "../../../../shared/helpers/SirenHelper";
import ProviderValueFactory from "../../../../shared/ProviderValueFactory";
import { Etablissement, Association } from "@api-subventions-asso/dto";
import { Document } from "@api-subventions-asso/dto/search/Document";
import StructureDto, {
    StructureDacDocumentDto,
    StructureEtablissementDto,
    StructureRepresentantLegalDto,
    StructureRibDto,
    StructureRnaDocumentDto
} from "../dto/StructureDto";
import { isValidDate } from "../../../../shared/helpers/DateHelper";

export default class ApiAssoDtoAdapter {
    static providerNameRna = "RNA";
    static providerNameLcaDocument = "Le Compte Asso";
    static providerNameSiren = "SIREN";

    static toAssociation(structure: StructureDto): Association[] {
        const associations: Association[] = [];
        const fromRNA = structure.identite.regime === "loi1901"; // Data come from rna
        const toDate = (stringDate: string) => {
            const [year, month, day] = stringDate.split("-").map(string => parseInt(string, 10));
            return new Date(Date.UTC(year, month - 1, day));
        };

        if (structure.identite.date_modif_rna) {
            const toRnaPvs = ProviderValueFactory.buildProviderValuesAdapter(
                this.providerNameRna,
                toDate(structure.identite.date_modif_rna)
            );
            const rnaAssociation: Association = {
                rna: toRnaPvs(structure.identite.id_rna),
                denomination_rna: toRnaPvs(structure.identite.nom),
                date_creation_rna: structure.identite.date_creat
                    ? toRnaPvs(toDate(structure.identite.date_creat))
                    : undefined,
                date_modification_rna: toRnaPvs(toDate(structure.identite.date_modif_rna)),
                objet_social: toRnaPvs(structure.activites.objet),
                code_objet_social_1: toRnaPvs(structure.activites.id_objet_social1),
                code_objet_social_2: toRnaPvs(structure.activites.id_objet_social2),
                adresse_siege_rna: fromRNA
                    ? toRnaPvs({
                          numero: structure.coordonnees.adresse_siege.num_voie,
                          type_voie: structure.coordonnees.adresse_siege.type_voie,
                          voie: structure.coordonnees.adresse_siege.voie,
                          code_postal: structure.coordonnees.adresse_siege.cp,
                          commune: structure.coordonnees.adresse_siege.commune
                      })
                    : undefined
            };
            associations.push(rnaAssociation);
        }

        if (structure.identite.date_modif_siren) {
            const toSirenPvs = ProviderValueFactory.buildProviderValuesAdapter(
                this.providerNameSiren,
                toDate(structure.identite.date_modif_siren)
            );
            const adresse =
                structure.coordonnees.adresse_siege_sirene ||
                (fromRNA ? undefined : structure.coordonnees.adresse_siege);
            const sirenAssociation: Association = {
                denomination_siren: structure.identite.nom_sirene
                    ? toSirenPvs(structure.identite.nom_sirene)
                    : toSirenPvs(structure.identite.nom),
                siren: toSirenPvs(structure.identite.id_siren),
                nic_siege: toSirenPvs(siretToNIC(structure.identite.id_siret_siege)),
                categorie_juridique: toSirenPvs(structure.identite.id_forme_juridique.toString()),
                date_creation_siren: structure.identite.date_creation_sirene
                    ? toSirenPvs(toDate(structure.identite.date_creation_sirene))
                    : undefined,
                date_modification_siren: toSirenPvs(toDate(structure.identite.date_modif_siren)),
                adresse_siege_siren: adresse
                    ? toSirenPvs({
                          numero: adresse.num_voie,
                          type_voie: adresse.type_voie,
                          voie: adresse.voie,
                          code_postal: adresse.cp,
                          commune: adresse.commune
                      })
                    : undefined,
                etablisements_siret: toSirenPvs(structure.etablissement.map(e => e.id_siret))
            };
            associations.push(sirenAssociation);
        }

        return associations;
    }

    static toEtablissement(
        etablissement: StructureEtablissementDto,
        ribs: StructureRibDto[],
        representantsLegaux: StructureRepresentantLegalDto[],
        dateModif: string
    ): Etablissement {
        const toDate = (stringDate: string) => {
            const [year, month, day] = stringDate.split("-").map(string => parseInt(string, 10));
            return new Date(Date.UTC(year, month - 1, day));
        };
        const toSirenPvs = ProviderValueFactory.buildProviderValuesAdapter(this.providerNameSiren, toDate(dateModif));
        const toLCAPvs = ProviderValueFactory.buildProviderValuesAdapter(
            this.providerNameLcaDocument,
            toDate(dateModif)
        );

        const toContact = (r: StructureRepresentantLegalDto) => ({
            nom: r.nom,
            prenom: r.prenom,
            civilite: r.civilitÃ©,
            telephone: r.telephone,
            email: r.courriel,
            role: r.fonction
        });

        return {
            siret: toSirenPvs(etablissement.id_siret),
            nic: toSirenPvs(siretToNIC(etablissement.id_siret)),
            ouvert: toSirenPvs(etablissement.actif),
            siege: toSirenPvs(etablissement.est_siege),
            adresse: toSirenPvs({
                numero: etablissement.adresse.num_voie,
                type_voie: etablissement.adresse.type_voie,
                voie: etablissement.adresse.voie,
                code_postal: etablissement.adresse.cp,
                commune: etablissement.adresse.commune
            }),
            information_banquaire: ribs
                .filter(rib => rib.id_siret === etablissement.id_siret)
                .map(rib => toLCAPvs({ iban: rib.iban, bic: rib.bic })),
            representants_legaux: representantsLegaux
                ? representantsLegaux
                      .filter(r => r.id_siret === etablissement.id_siret)
                      .map(r => toLCAPvs(toContact(r)))
                : undefined,
            contacts: representantsLegaux
                ? representantsLegaux
                      .filter(r => r.id_siret === etablissement.id_siret)
                      .map(r => toLCAPvs(toContact(r)))
                : undefined
        };
    }

    static rnaDocumentToDocument(rnaDocument: StructureRnaDocumentDto): Document {
        let date = new Date(Date.UTC(rnaDocument.annee as number, 0));
        // DTO expect date so we use 1970 as a hack to know that the date is not defined
        if (!isValidDate(date)) date = new Date(Date.UTC(1970, 0));
        else if (rnaDocument.time) date.setTime(date.getTime() + rnaDocument.time);

        const toRnaPv = ProviderValueFactory.buildProviderValueAdapter(this.providerNameRna, date);
        return {
            nom: toRnaPv(`${rnaDocument.lib_sous_type} - ${rnaDocument.id}`),
            type: toRnaPv(rnaDocument.sous_type),
            url: toRnaPv(rnaDocument.url),
            __meta__: {}
        };
    }

    static dacDocumentToDocument(dacDocument: StructureDacDocumentDto): Document {
        const isoDate = new Date(dacDocument.time_depot);
        const toLCAPv = ProviderValueFactory.buildProviderValueAdapter(
            this.providerNameLcaDocument,
            new Date(
                Date.UTC(
                    isoDate.getFullYear(),
                    isoDate.getMonth(),
                    isoDate.getDate(),
                    isoDate.getHours(),
                    isoDate.getMinutes()
                )
            )
        );

        return {
            nom: toLCAPv(dacDocument.nom),
            type: toLCAPv(dacDocument.meta.type),
            url: toLCAPv(dacDocument.url),
            __meta__: {
                siret: String(dacDocument.meta.id_siret)
            }
        };
    }

    static dacDocumentToRib(rib: StructureDacDocumentDto): Document {
        const isoDate = new Date(rib.time_depot);

        const toRnaPv = ProviderValueFactory.buildProviderValueAdapter(this.providerNameRna, isoDate);
        return {
            nom: toRnaPv(rib.meta.iban || rib.nom),
            type: toRnaPv("RIB"),
            url: toRnaPv(rib.url),
            __meta__: {
                siret: String(rib.meta.id_siret)
            }
        };
    }
}
