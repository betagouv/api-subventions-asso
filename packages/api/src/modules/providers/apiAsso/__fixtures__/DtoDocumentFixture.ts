import ApiAssoDtoAdapter from "../adapters/ApiAssoDtoAdapter";

export const RnaDtoDocument = {
    __meta__: {},
    nom: {
        last_update: new Date("2021-01-19T18:42:37.171Z"),
        provider: ApiAssoDtoAdapter.providerNameRna,
        type: "string",
        value: "PV - ididididid"
    },
    type: {
        last_update: new Date("2021-01-19T18:42:37.171Z"),
        provider: ApiAssoDtoAdapter.providerNameRna,
        type: "string",
        value: "PV"
    },
    url: {
        last_update: new Date("2021-01-19T18:42:37.171Z"),
        provider: ApiAssoDtoAdapter.providerNameRna,
        type: "string",
        value: "/fake/url"
    }
};

export const DacSiret = "50922194100000";

export const DacDtoDocument = {
    __meta__: {
        siret: DacSiret
    },
    nom: {
        last_update: new Date("2021-06-18T12:02:00.000Z"),
        provider: ApiAssoDtoAdapter.providerNameLcaDocument,
        type: "string",
        value: "nom fake"
    },
    type: {
        last_update: new Date("2021-06-18T12:02:00.000Z"),
        provider: ApiAssoDtoAdapter.providerNameLcaDocument,
        type: "string",
        value: "BPA"
    },
    url: {
        last_update: new Date("2021-06-18T12:02:00.000Z"),
        provider: ApiAssoDtoAdapter.providerNameLcaDocument,
        type: "string",
        value: "/fake/url"
    }
};

export const DtoDocument = [RnaDtoDocument, DacDtoDocument];
