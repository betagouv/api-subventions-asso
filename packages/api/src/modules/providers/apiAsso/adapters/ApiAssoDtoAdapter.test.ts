import ApiAssoDtoAdapter from "./ApiAssoDtoAdapter";
import { StructureRepresentantLegalDto } from "../dto/StructureDto";
import {
    fixtureAsso,
    fixtureEtablissements,
    fixtureRepresentantLegal,
    fixtureRib
} from "../__fixtures__/ApiAssoStructureFixture";
import { DacDtoDocument, RnaDtoDocument } from "../__fixtures__/DtoDocumentFixture";
import { ApiAssoDocumentFixture } from "../__fixtures__/ApiAssoDocumentFixture";
import { sirenStructureFixture } from "../__fixtures__/SirenStructureFixture";
import { rnaStructureFixture } from "../__fixtures__/RnaStructureFixture";

describe("ApiAssoDtoAdapter", () => {
    describe("toEtablissement", () => {
        it("should return etablissement with rib", () => {
            const actual = ApiAssoDtoAdapter.toEtablissement(
                fixtureEtablissements[0],
                fixtureRib,
                [],
                fixtureAsso.identite.date_modif_siren
            );

            expect(actual).toMatchSnapshot();
        });

        it("should return etablissement with contact", () => {
            const actual = ApiAssoDtoAdapter.toEtablissement(
                fixtureEtablissements[0],
                [],
                fixtureRepresentantLegal,
                fixtureAsso.identite.date_modif_siren
            );

            expect(actual).toMatchSnapshot();
        });

        it("should return etablissement without rib", () => {
            const actual = ApiAssoDtoAdapter.toEtablissement(
                fixtureEtablissements[1],
                fixtureRib,
                [],
                fixtureAsso.identite.date_modif_siren
            );

            expect(actual).toMatchSnapshot();
        });

        it("should return etablissement without contact", () => {
            const actual = ApiAssoDtoAdapter.toEtablissement(
                fixtureEtablissements[1],
                [],
                undefined as unknown as StructureRepresentantLegalDto[],
                fixtureAsso.identite.date_modif_siren
            );

            expect(actual).toMatchSnapshot();
        });
    });

    describe("rnaDocumentToDocument", () => {
        it("should return StructureRnaDocumentDto", () => {
            const expected = RnaDtoDocument;
            const actual = ApiAssoDtoAdapter.rnaDocumentToDocument(
                ApiAssoDocumentFixture.asso.documents.document_rna[0]
            );
            expect(actual).toEqual(expected);
        });

        it("should set date to 01/01/1970 if year is not define", () => {
            const document_rna = {
                ...ApiAssoDocumentFixture.asso.documents.document_rna[0],
                annee: undefined
            };
            const actual = ApiAssoDtoAdapter.rnaDocumentToDocument(document_rna);
            expect(actual).toMatchSnapshot();
        });

        it("should set date to 01/01 of year if time is not define", () => {
            const document_rna = {
                ...ApiAssoDocumentFixture.asso.documents.document_rna[0],
                time: undefined
            };
            const actual = ApiAssoDtoAdapter.rnaDocumentToDocument(document_rna);
            expect(actual).toMatchSnapshot();
        });
    });

    describe("dacDocumentToDocument", () => {
        it("should return StructureDacDocumentDto", () => {
            const expected = DacDtoDocument;
            const actual = ApiAssoDtoAdapter.dacDocumentToDocument(
                ApiAssoDocumentFixture.asso.documents.document_dac[0]
            );
            expect(actual).toEqual(expected);
        });
    });

    describe("sirenStructureToAssociation", () => {
        it("should transform to association", () => {
            expect(ApiAssoDtoAdapter.sirenStructureToAssociation(sirenStructureFixture)).toMatchSnapshot();
        });
    });

    describe("rnaStructureToAssociation", () => {
        it("should transform to association", () => {
            expect(ApiAssoDtoAdapter.rnaStructureToAssociation(rnaStructureFixture)).toMatchSnapshot();
        });
    });

    describe("apiDateToDate", () => {
        it("should return valid date", () => {
            const actual = ApiAssoDtoAdapter.apiDateToDate("2022-12-23");

            expect(actual).toEqual(new Date(Date.UTC(2022, 11, 23)));
        });
    });
});
