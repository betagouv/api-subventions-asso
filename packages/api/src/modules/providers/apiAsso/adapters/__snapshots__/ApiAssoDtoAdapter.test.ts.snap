// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiAssoDtoAdapter rnaDocumentToDocument should set date to 01/01 of year if time is not define 1`] = `
Object {
  "__meta__": Object {},
  "nom": Object {
    "last_update": 2021-01-01T00:00:00.000Z,
    "provider": "RNA",
    "type": "string",
    "value": "PV - ididididid",
  },
  "type": Object {
    "last_update": 2021-01-01T00:00:00.000Z,
    "provider": "RNA",
    "type": "string",
    "value": "PV",
  },
  "url": Object {
    "last_update": 2021-01-01T00:00:00.000Z,
    "provider": "RNA",
    "type": "string",
    "value": "/fake/url",
  },
}
`;

exports[`ApiAssoDtoAdapter rnaDocumentToDocument should set date to 01/01/1970 if year is not define 1`] = `
Object {
  "__meta__": Object {},
  "nom": Object {
    "last_update": 1970-01-01T00:00:00.000Z,
    "provider": "RNA",
    "type": "string",
    "value": "PV - ididididid",
  },
  "type": Object {
    "last_update": 1970-01-01T00:00:00.000Z,
    "provider": "RNA",
    "type": "string",
    "value": "PV",
  },
  "url": Object {
    "last_update": 1970-01-01T00:00:00.000Z,
    "provider": "RNA",
    "type": "string",
    "value": "/fake/url",
  },
}
`;

exports[`ApiAssoDtoAdapter toAssociation should return rna association 1`] = `
Object {
  "adresse_siege_rna": undefined,
  "code_objet_social_1": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "code_objet_social_2": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "date_creation_rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "object",
      "value": 1994-07-04T00:00:00.000Z,
    },
  ],
  "date_modification_rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "object",
      "value": 2018-01-01T00:00:00.000Z,
    },
  ],
  "denomination_rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "string",
      "value": "TEST",
    },
  ],
  "objet_social": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "string",
      "value": "W00000000",
    },
  ],
}
`;

exports[`ApiAssoDtoAdapter toAssociation should return rna association with adresse 1`] = `
Object {
  "adresse_siege_rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "object",
      "value": Object {
        "code_postal": undefined,
        "commune": "paris",
        "numero": undefined,
        "type_voie": undefined,
        "voie": undefined,
      },
    },
  ],
  "code_objet_social_1": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "code_objet_social_2": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "date_creation_rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "object",
      "value": 1994-07-04T00:00:00.000Z,
    },
  ],
  "date_modification_rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "object",
      "value": 2018-01-01T00:00:00.000Z,
    },
  ],
  "denomination_rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "string",
      "value": "TEST",
    },
  ],
  "objet_social": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "string",
      "value": "W00000000",
    },
  ],
}
`;

exports[`ApiAssoDtoAdapter toAssociation should return rna association with date creation 1`] = `
Object {
  "adresse_siege_rna": undefined,
  "code_objet_social_1": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "code_objet_social_2": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "date_creation_rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "object",
      "value": 1906-07-15T00:00:00.000Z,
    },
  ],
  "date_modification_rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "object",
      "value": 2018-01-01T00:00:00.000Z,
    },
  ],
  "denomination_rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "string",
      "value": "TEST",
    },
  ],
  "objet_social": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "rna": Array [
    Object {
      "last_update": 2018-01-01T00:00:00.000Z,
      "provider": "RNA",
      "type": "string",
      "value": "W00000000",
    },
  ],
}
`;

exports[`ApiAssoDtoAdapter toAssociation should return siren association 1`] = `
Object {
  "adresse_siege_siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": Object {
        "code_postal": undefined,
        "commune": "paris",
        "numero": undefined,
        "type_voie": undefined,
        "voie": undefined,
      },
    },
  ],
  "categorie_juridique": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "9220",
    },
  ],
  "date_creation_siren": undefined,
  "date_modification_siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": 2010-01-01T00:00:00.000Z,
    },
  ],
  "denomination_siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "TEST SIREN 2",
    },
  ],
  "etablisements_siret": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": Array [
        "50922194100000",
        "50922194100001",
      ],
    },
  ],
  "nic_siege": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "0000",
    },
  ],
  "siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "509221941",
    },
  ],
}
`;

exports[`ApiAssoDtoAdapter toAssociation should return siren association with date creation 1`] = `
Object {
  "adresse_siege_siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": Object {
        "code_postal": undefined,
        "commune": "paris",
        "numero": undefined,
        "type_voie": undefined,
        "voie": undefined,
      },
    },
  ],
  "categorie_juridique": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "9220",
    },
  ],
  "date_creation_siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": 1906-07-15T00:00:00.000Z,
    },
  ],
  "date_modification_siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": 2010-01-01T00:00:00.000Z,
    },
  ],
  "denomination_siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "TEST SIREN 2",
    },
  ],
  "etablisements_siret": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": Array [
        "50922194100000",
        "50922194100001",
      ],
    },
  ],
  "nic_siege": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "0000",
    },
  ],
  "siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "509221941",
    },
  ],
}
`;

exports[`ApiAssoDtoAdapter toAssociation should return siren association without nom siren 1`] = `
Object {
  "adresse_siege_siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": Object {
        "code_postal": undefined,
        "commune": "paris",
        "numero": undefined,
        "type_voie": undefined,
        "voie": undefined,
      },
    },
  ],
  "categorie_juridique": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "9220",
    },
  ],
  "date_creation_siren": undefined,
  "date_modification_siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": 2010-01-01T00:00:00.000Z,
    },
  ],
  "denomination_siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "TEST",
    },
  ],
  "etablisements_siret": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": Array [
        "50922194100000",
        "50922194100001",
      ],
    },
  ],
  "nic_siege": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "0000",
    },
  ],
  "siren": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "509221941",
    },
  ],
}
`;

exports[`ApiAssoDtoAdapter toEtablissement should return etablissement with contact 1`] = `
Object {
  "adresse": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": Object {
        "code_postal": undefined,
        "commune": undefined,
        "numero": undefined,
        "type_voie": undefined,
        "voie": undefined,
      },
    },
  ],
  "contacts": Array [
    Array [
      Object {
        "last_update": 2010-01-01T00:00:00.000Z,
        "provider": "Le Compte Asso",
        "type": "object",
        "value": Object {
          "civilite": undefined,
          "email": undefined,
          "nom": "Jedusor",
          "prenom": "Tom",
          "role": undefined,
          "telephone": undefined,
        },
      },
    ],
  ],
  "information_banquaire": Array [],
  "nic": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "00000",
    },
  ],
  "ouvert": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "representants_legaux": Array [
    Array [
      Object {
        "last_update": 2010-01-01T00:00:00.000Z,
        "provider": "Le Compte Asso",
        "type": "object",
        "value": Object {
          "civilite": undefined,
          "email": undefined,
          "nom": "Jedusor",
          "prenom": "Tom",
          "role": undefined,
          "telephone": undefined,
        },
      },
    ],
  ],
  "siege": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "siret": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "50922194100000",
    },
  ],
}
`;

exports[`ApiAssoDtoAdapter toEtablissement should return etablissement with rib 1`] = `
Object {
  "adresse": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": Object {
        "code_postal": undefined,
        "commune": undefined,
        "numero": undefined,
        "type_voie": undefined,
        "voie": undefined,
      },
    },
  ],
  "contacts": Array [],
  "information_banquaire": Array [
    Array [
      Object {
        "last_update": 2010-01-01T00:00:00.000Z,
        "provider": "Le Compte Asso",
        "type": "object",
        "value": Object {
          "bic": "BIC_TEST",
          "iban": "TEST",
        },
      },
    ],
    Array [
      Object {
        "last_update": 2010-01-01T00:00:00.000Z,
        "provider": "Le Compte Asso",
        "type": "object",
        "value": Object {
          "bic": "BIC_TEST 2",
          "iban": "TEST 2",
        },
      },
    ],
  ],
  "nic": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "00000",
    },
  ],
  "ouvert": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "representants_legaux": Array [],
  "siege": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "siret": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "50922194100000",
    },
  ],
}
`;

exports[`ApiAssoDtoAdapter toEtablissement should return etablissement without contact 1`] = `
Object {
  "adresse": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": Object {
        "code_postal": undefined,
        "commune": undefined,
        "numero": undefined,
        "type_voie": undefined,
        "voie": undefined,
      },
    },
  ],
  "contacts": undefined,
  "information_banquaire": Array [],
  "nic": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "00001",
    },
  ],
  "ouvert": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "representants_legaux": undefined,
  "siege": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "siret": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "50922194100001",
    },
  ],
}
`;

exports[`ApiAssoDtoAdapter toEtablissement should return etablissement without rib 1`] = `
Object {
  "adresse": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "object",
      "value": Object {
        "code_postal": undefined,
        "commune": undefined,
        "numero": undefined,
        "type_voie": undefined,
        "voie": undefined,
      },
    },
  ],
  "contacts": Array [],
  "information_banquaire": Array [],
  "nic": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "00001",
    },
  ],
  "ouvert": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "representants_legaux": Array [],
  "siege": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "undefined",
      "value": undefined,
    },
  ],
  "siret": Array [
    Object {
      "last_update": 2010-01-01T00:00:00.000Z,
      "provider": "SIREN",
      "type": "string",
      "value": "50922194100001",
    },
  ],
}
`;
